### ISSUE TITLE
共享对象 `Style` 的样式数据，减少内存消耗并提升控件构建速度

### ISSUE THREAD

**Issue by Xiaokang2022** [Xiaokang2022/maliang#129]
### Feature Description

按照目前 `maliang` 的设计，每个类 `Style` 在实例化的时候，都会将类变量 `light` 和 `dark` 深拷贝一次作为自己的实例变量。此方法不仅会消耗大量的内存，而且还会增大控件的构建时间。

**考虑**：当对象 `Style` 的 `light` 和 `dark` 没有被编辑的时候，直接采用类变量 `light` 和 `dark` 的引用，被编辑之后再深拷贝原数据以分离出来。

**问题**：目前编辑 `Style` 对象数据的方法有两套，这两套方法可能不能同时实现该 Issue 的功能。

- [X] 第一套方法：

```python
widget.style.set(bg="red")
```

- [x] 第二套方法：

```python
widget.style.style[0]["normal"] = {"fill": "red"}
```

优先尝试为第一套方法实现该功能。

### Possible Implementation

或许可以采用设计模式 “享元模式” 或采用 “写时复制” 等技术。

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by Xiaokang2022** [comment]
已在提交 https://github.com/Xiaokang2022/maliang/commit/ea74ef262369c4b5f0d0fed296b4492a84bd349c 中实现对第一套方法的优化

**Comment by Xiaokang2022** [comment]
在提交 https://github.com/Xiaokang2022/maliang/commit/65d6921e0545d4f84e5e124479bc61cbc97bf9ec 中给类 `Style` 添加了方法 `detach` 来主动生成实例变量。对于上述的第二套方法，可以通过链式调用的方式解决：

```python
widget.style.style.detach()[0]["normal"] = {"fill": "red"}
```

且对于当前控件，仅需调用一次，后续无需再次链式调用方法 `detach`。

**Comment by Xiaokang2022** [comment]
经测试，重复构建相同样式的控件，内存降低约 $$5.5\\%$$。

**Comment by Xiaokang2022** [comment]
此更新跟随版本 `3.1.4` 发布
